src/oic/utils/clientdb.py

```diff
@@ -1,22 +1,134 @@
-import json
-
+"""Client managament databases."""
 import requests
 from six.moves.urllib.parse import quote
+from six.moves.urllib.parse import urljoin
 
 from oic.oauth2.exception import NoClientInfoReceivedError
 
 
-class MDQClient(object):
+class BaseClientDatabase(object):
```

Comment:
Would it make any sense to use abc.ABCMeta here to declare the baseclass and get rid of the NotImplementedErrors in exchange for @abstractmethod decorators?

Reply:
I never used this approach, so I have no experience with it. I will look into this on Monday.